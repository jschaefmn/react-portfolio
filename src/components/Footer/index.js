import React from 'react';

function Footer() {
  return (
    <footer>
      <h2>Connect with me</h2>
      <ul>
        <li>
          &copy; {new Date().getFullYear()} by Jacob Schaefbauer
        </li>
        <li>
          <a href="https://github.com/jschaefmn" target="_blank" rel="noreferrer">GitHub</a>
        </li>
        <li>
          <a href="https://www.linkedin.com/in/jacob-schaefbauer-095107140/" target="_blank" rel="noreferrer">LinkedIn</a>
        </li>
        <li>
          <a href="https://twitter.com/MrCoolJakey">Twitter</a>
        </li>
      </ul>
    </footer>
  );
}

export default Footer;