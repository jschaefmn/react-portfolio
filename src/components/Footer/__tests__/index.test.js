import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '..';

afterEach(cleanup);

describe('Footer component', () => {
  it('renders', () => {
    render(<Footer />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});

// test links in contact section
test('renders footer links', () => {
  render(<Footer />);
  const githubLink = screen.getByText(/GitHub/i);
  const linkedinLink = screen.getByText(/LinkedIn/i);
  const twitterLink = screen.getByText(/Twitter/i);
  expect(githubLink).toBeInTheDocument();
  expect(linkedinLink).toBeInTheDocument();
  expect(twitterLink).toBeInTheDocument();
});