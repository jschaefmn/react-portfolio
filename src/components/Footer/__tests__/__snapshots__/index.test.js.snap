// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer>
    <h2>
      Connect with me
    </h2>
    <ul>
      <li>
        Â© 2022 by Jacob Schaefbauer
      </li>
      <li>
        <a
          href="https://github.com/jschaefmn"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/jacob-schaefbauer-095107140/"
          rel="noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/MrCoolJakey"
        >
          Twitter
        </a>
      </li>
    </ul>
  </footer>
</DocumentFragment>
`;
